[Feature] Bump up Qt requirements to 5.12

We're using 5.7.1 as the base Qt version. This release is from December 2016. 

Using a relatively older version has proven to be useful in terms of compatibility with older distributions.

However.... Newer versions have bugfixes and new fancy features. More specifically, since `5.12` which released recently, we can now use ECMAScript7. Arrow functions, somewhat proper OOP, variable scope control, list goes on.

I propose to use Qt 5.12 for the upcoming 0.14 release of the GUI.

qt.5.9.1 cannot build this am getting this error

../src/libwalletqt/WalletManager.cpp:51:16: error: 'virtual void WalletPassphraseListenerImpl::onDeviceButtonPressed()' marked 'override', but does not override
   virtual void onDeviceButtonPressed() override
                ^
../src/libwalletqt/WalletManager.cpp:56:16: error: 'virtual void WalletPassphraseListenerImpl::onSetWallet(Monero::Wallet*)' marked 'override', but does not override
   virtual void onSetWallet(Monero::Wallet * wallet) override
                ^

--- a/README.md
+++ b/README.md
@@ -219,9 +219,9 @@ Packaging for your favorite distribution would be a welcome contribution!
 
 2. Install Qt:
 
-  *Note*: The Qt 5.9.7 or newer requirement makes **some** distributions (mostly based on Debian, like Ubuntu 16.x or Linux Mint 18.x) obsolete due to their repositories containing an older Qt version.
+  *Note*: The Qt 5.12 or newer requirement makes **some** distributions (mostly based on Debian, like Ubuntu 16.x or Linux Mint 18.x) obsolete due to their repositories containing an older Qt version.
 
- The recommended way is to install 5.9.7 from the [official Qt installer](https://www.qt.io/download-qt-installer) or [compiling it yourself](https://wiki.qt.io/Install_Qt_5_on_Ubuntu). This ensures you have the correct version. Higher versions *can* work but as it differs from our production build target, slight differences may occur.
+ The recommended way is to install 5.12 or newer from the [official Qt installer](https://www.qt.io/download-qt-installer) or [compiling it yourself](https://wiki.qt.io/Install_Qt_5_on_Ubuntu). This ensures you have the correct version. Higher versions *can* work but as it differs from our production build target, slight differences may occur.
 
 The following instructions will fetch Qt from your distribution's repositories instead. Take note of what version it installs. Your mileage may vary.
 
@@ -277,7 +277,7 @@ The executable can be found in the build/release/bin folder.
 
 4. Install Qt:
 
-  `brew install qt5`  (or download QT 5.9.7+ from [qt.io](https://www.qt.io/download-open-source/))
+  `brew install qt5`  (or download QT 5.12+ from [qt.io](https://www.qt.io/download-open-source/))
 
 5. Grab an up-to-date copy of the monero-gui repository
 

